app/livechat/client/views/app/tabbar/visitorEdit.js

```diff
@@ -93,13 +92,11 @@ Template.visitorEdit.onCreated(function() {
 	});
 
 	const uid = Meteor.userId();
-	this.subscribe('livechat:departmentAgents', null, uid, () => {
-		const departments = [];
-		LivechatDepartmentAgents.find({ agentId: uid }).forEach((dept) => {
-			departments.push(dept.departmentId);
-		});
-		this.agentDepartments.set(departments);
+	const { departments } = await APIClient.v1.get(`livechat/agent/${ uid }/departments`);
+	departments.forEach((dept) => {
+		departments.push(dept.departmentId);
```

Comment:
You're using the same const you get from the endpoint to add the `id` of each department, so the result will look like this:

```(6) [{…}, {…}, {…}, "mHEd86Qzg2YfvdBim", "DQY5Tg62Qnzc4qaqr", "mRRDePBQJmePiTreM"]
0: {_id: "jFJM3jgpsXKf8jCF2", agentId: "o8WnaGnQrCAaJ7xGB", departmentId: "mHEd86Qzg2YfvdBim", _updatedAt: "2019-10-11T21:24:18.949Z", count: 0, …}
1: {_id: "86KoB5XgdEXfbBkeb", agentId: "o8WnaGnQrCAaJ7xGB", departmentId: "DQY5Tg62Qnzc4qaqr", _updatedAt: "2019-10-18T18:58:26.736Z", count: 3, …}
2: {_id: "cHSBiPzmbGqG9iJTL", agentId: "o8WnaGnQrCAaJ7xGB", departmentId: "mRRDePBQJmePiTreM", _updatedAt: "2019-10-18T18:58:41.044Z", count: 0, …}
3: "mHEd86Qzg2YfvdBim"
4: "DQY5Tg62Qnzc4qaqr"
5: "mRRDePBQJmePiTreM"
```

Please, use a new const(array) to store the `id` of the departments.